src/Servers/Kestrel/shared/test/TransportTestHelpers/TestServer.cs

```diff
@@ -130,9 +135,11 @@ public Task StopAsync(CancellationToken token = default)
             return _host.StopAsync(token);
         }
 
-        public void Dispose()
+        public async ValueTask DisposeAsync()
         {
-            _host.Dispose();
+            await _host.StopAsync();
+            // The concrete Host implements IAsyncDisposable
+            await ((IAsyncDisposable)_host).ConfigureAwait(false).DisposeAsync();
```

Comment:
If we're going to call ConfigureAwait(false) here (which makes sense to me b/c of the xunit SyncContext), we should do the same for _host.StopAsync().

Nit: The IAsyncEnumerable ConfigureAwait extension method is meant for when you're disposing the IAsyncDisposable with `await using` (https://github.com/dotnet/runtime/issues/27857). We should probably just call DisposeAsync().ConfigureAwait(false) the normal way here. I know we already do it this way in the other TestServer. I must have missed that when it got added.